Provisioning: Makes file the default dashboard provisioner type

bergquist opened the pull request:
closes https://github.com/grafana/grafana/issues/23111

bergquist committed:
convert convey tests to stdlib tests

bergquist committed:
use file as default type

bergquist committed:
adds docs about default value for type

marefr reviewed (approved):
LGTM, but hard to see what docs changes you've made due to a lot of formatting changes

bergquist commented:
@marefr yes. vscode went crazy on those files. I'll make sure to split that into multiple commits next time.

bergquist commented:
I've removed "Required" on the folder setting (cause it's not, it will put dashboards in the root folder if it's missing) and added a note about file being the default for `type`.

aknuds1 commented on pkg/services/provisioning/dashboards/config_reader_test.go:
```diff
@@ -4,87 +4,99 @@ import (
+		t.Run("default values should be applied", func(t *testing.T) {
+			cfgProvider := configReader{path: appliedDefaults, log: logger}
+			cfg, err := cfgProvider.readConfig()
+			require.Nil(t, err)
```
```suggestion
			require.NoError(t, err)
```

aknuds1 commented on pkg/services/provisioning/dashboards/config_reader_test.go:
```diff
@@ -4,87 +4,99 @@ import (
 			cfg, err := cfgProvider.readConfig()
 			_ = os.Unsetenv("TEST_VAR")
-			So(err, ShouldBeNil)
+			require.Nil(t, err)
```
```suggestion
			require.NoError(t, err)
```

aknuds1 commented on pkg/services/provisioning/dashboards/config_reader_test.go:
```diff
@@ -4,87 +4,99 @@ import (
 			cfgProvider := configReader{path: oldVersion, log: logger}
 			cfg, err := cfgProvider.readConfig()
-			So(err, ShouldBeNil)
+			require.Nil(t, err)
```
```suggestion
			require.NoError(t, err)
```

aknuds1 reviewed (approved):
LGTM, just left some suggestions.

bergquist committed:
Apply suggestions from code review

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>
